internal/metastore/kv/rootcoord/kv_catalog.go

```diff
@@ -1439,6 +1466,113 @@ func (kc *Catalog) RestoreRBAC(ctx context.Context, tenant string, meta *milvusp
 	return err
 }
 
+func (kc *Catalog) GetPrivilegeGroup(ctx context.Context, groupName string) ([]*milvuspb.PrivilegeEntity, error) {
+	k := BuildPrivilegeGroupkey(groupName)
+	val, err := kc.Txn.Load(k)
+	if err != nil {
+		if errors.Is(err, merr.ErrIoKeyNotFound) {
+			return nil, fmt.Errorf("privilege group [%s] does not exist", groupName)
+		}
+		log.Error("failed to load privilege group", zap.String("group", groupName), zap.Error(err))
+		return nil, err
+	}
+	privGroupInfo := &milvuspb.PrivilegeGroupInfo{}
+	err = proto.Unmarshal([]byte(val), privGroupInfo)
+	if err != nil {
+		log.Error("failed to unmarshal privilege group info", zap.Error(err))
+		return nil, err
+	}
+	return privGroupInfo.Privileges, nil
+}
+
+func (kc *Catalog) DropPrivilegeGroup(ctx context.Context, groupName string) error {
+	// check if the group is used by any role
+	roles, err := kc.ListRole(ctx, util.DefaultTenant, nil, false)
+	if err != nil {
+		return err
+	}
+
+	roleEntity := lo.FilterMap(roles, func(entity *milvuspb.RoleResult, _ int) (*milvuspb.RoleEntity, bool) {
+		roleName := entity.GetRole().GetName()
+		if roleName == util.RoleAdmin || roleName == util.RolePublic {
+			return nil, false
+		}
+
+		return entity.GetRole(), true
+	})
+
+	for _, role := range roleEntity {
+		grants, err := kc.ListGrant(ctx, util.DefaultTenant, &milvuspb.GrantEntity{
+			Role:   role,
+			DbName: util.AnyWord,
+		})
+		if err != nil {
+			return err
+		}
+		for _, grant := range grants {
+			if grant.Grantor.Privilege.Name == groupName {
+				return errors.Newf("privilege group [%s] is used by role [%s], Use REVOKE API to revoke it first", groupName, role.GetName())
+			}
+		}
```

Comment:
prefer to do the judge logic in root_coord.go, catalog should only keeps logic about reading/writing meta to etcd